.github/workflows/publish.yml

```
@@ -0,0 +1,75 @@
+name: Publish
+
+on:
+  push:
+    branches: ["main"]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  provider-development-kit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: setup-git-credentials
+        uses: fusion-engineering/setup-git-credentials@v2.0.6
+        with:
+          credentials: "https://fiberplanebot:${{ secrets.PRIVATE_GITHUB_TOKEN }}@github.com/"
+
+      - name: Cache Rust dependencies
+        uses: actions/cache@v2
```

```suggestion
        uses: actions/cache@v3
```

(same for the other job)